addons/account_check_printing/models/account_payment.py

```diff
@@ -61,8 +61,9 @@ def _constrains_check_number(self):
                AND payment.id IN %(ids)s
                AND move.state = 'posted'
                AND other_move.state = 'posted'
+               AND payment.check_number IS NOT NULL
```

Comment:
`AND payment.check_number IS NOT NULL` --> `AND other_move.check_number IS NOT NULL`

Reply:
Sorry, I even didn't get you what change are you requesting...

Let me try using the `Insert Suggestion` feature
 - ![Screen Shot 2022-01-19 at 18 34 48](https://user-images.githubusercontent.com/6644187/150240862-461b31c3-3ad1-4baa-ab17-5c7b23e7a9b9.png)

What change are you requesting me?

Option 1
```suggestion

```

Option 2
```suggestion
               AND other_payment.check_number IS NOT NULL
```

Option 3
```suggestion
               AND other_payment.check_number IS NOT NULL
               AND payment.check_number IS NOT NULL
```

Let me share you the result of each query:

| Where {} IS NOT NULL | Execution time (ms) | Option |
|---|---|---|
| original query w/o IS NOT NULL |  1855.064 |  option 1 |
| other_payment.check_number | 1792.995 |  option 2 |
| payment.check_number AND other_payment.check_number |  0.075 |  option 3 |
| payment.check_number | 0.078 |  without changes in this PR (current commit) |

Reply:
@odony what do you think?

Reply:
You are right, the best solution is option 3.
`other_payment.check_number IS NOT NULL` is mandatory
`payment.check_number IS NOT NULL` is optional because this condition is already in `payment.id IN payment_checks.ids`. But it help the SQL engine (first it select NOT NULL row (easy), then select with IN (not easy)).

Reply:
I got it

Thank you!

I have already applied the option 3
  - `payment.check_number AND other_payment.check_number`

Reply:
Hi @fmdl 

Something else to do from my part here?